local-cli/runAndroid/runAndroid.js

```diff
@@ -123,15 +123,28 @@ function buildAndRun(args, reject) {
 }
 
 function startServerInNewWindow() {
+  var yargV = require('yargs').argv;
+
   const launchPackagerScript = path.resolve(
     __dirname, '..', '..', 'packager', 'launchPackager.command'
   );
 
   if (process.platform === 'darwin') {
-    child_process.spawnSync('open', [launchPackagerScript]);
+    if (!yargV.open) {
+      return child_process.spawnSync('open', [launchPackagerScript]);
+    }
+    return child_process.spawnSync('open', ['-a', yargV.open, launchPackagerScript]);
+
   } else if (process.platform === 'linux') {
-    child_process.spawn(
-      'xterm',
+    if (!yargV.open){
```

Comment:
Same nit, except `-e sh` is specific to `xterm`. You might want to do something like:

```js
const term = yargV.term ? yargV.term : 'xterm';
const termArgs = (!yargV.termArgs && term === 'xterm') ? ['-e', 'sh'] : yargV.termArgs;
return child_process.spawn(term, termArgs.concat([launchPackagerScript]), {detached: true});
```

Please make sure to document these args as well. You would also need to declare `termArgs` in your `yargs` invocation.